packages/flutter/lib/src/material/icon_button.dart

```diff
@@ -341,6 +391,29 @@ class IconButton extends StatelessWidget {
   /// Null by default.
   final ButtonStyle? style;
 
+  /// The optional selection state of the icon button.
+  ///
+  /// If this property is null, the button will be a non-toggle button; otherwise,
+  /// the button will be a toggle button with the corresponding selection state.
+  ///
```

Comment:
```suggestion
  /// If this property is null, the button will behave as a normal push button,
  /// otherwise, the button will toggle between showing [icon] and [selectedIcon] 
  /// based on the value of [isSelected]. If true, it will show [selectedIcon], 
  /// if false it will show [icon]. 
  ///
```